packages/@aws-cdk/aws-apigatewayv2/lib/http/api.ts

```diff
@@ -457,12 +481,31 @@ export class HttpApi extends HttpApiBase {
    */
   public addRoutes(options: AddRoutesOptions): HttpRoute[] {
     const methods = options.methods ?? [HttpMethod.ANY];
-    return methods.map((method) => new HttpRoute(this, `${method}${options.path}`, {
-      httpApi: this,
-      routeKey: HttpRouteKey.with(options.path, method),
-      integration: options.integration,
-      authorizer: options.authorizer,
-      authorizationScopes: options.authorizationScopes,
-    }));
+    return methods.map((method) => {
+      let authorizationScopes = this.authorizationScopes;
+
+      if (Array.isArray(options.authorizationScopes)) {
+        authorizationScopes = options.authorizationScopes;
+      }
+
+      return new HttpRoute(this, `${method}${options.path}`, {
+        httpApi: this,
+        routeKey: HttpRouteKey.with(options.path, method),
+        integration: options.integration,
+        authorizer: options.authorizer ?? this.authorizer,
+        authorizationScopes,
```

Comment:
I would prefer we set this to `undefined` if the array is empty? Basically either non-empty array or undefined.

Reply:
Sure, made the change in the Route instead - https://github.com/aws/aws-cdk/pull/13172/commits/32aa124c5378a1d2c89b221f2d210e89b970f4d6#diff-7d9309e9aabf66aa9e84b4d545664646e6f7a44004598bcc6939ffe21eced3a4R161